clients/client-java/src/main/java/com/datastrato/gravitino/client/api/SupportsCatalogs.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.client.api;
+
+import com.datastrato.gravitino.Catalog;
+import com.datastrato.gravitino.CatalogChange;
+import com.datastrato.gravitino.CatalogProvider;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.annotation.Evolving;
+import com.datastrato.gravitino.exceptions.CatalogAlreadyExistsException;
+import com.datastrato.gravitino.exceptions.NoSuchCatalogException;
+import com.datastrato.gravitino.exceptions.NoSuchMetalakeException;
+import java.util.Map;
+
+/**
+ * Client interface for supporting catalogs. It includes methods for listing, loading, creating,
+ * altering and dropping catalogs. This class is forked from {@link
+ * com.datastrato.gravitino.SupportsCatalogs}, which was designed for Gravitino server side. The *
+ * difference is that this interface uses simple types instead of complex types in the method *
+ * signatures, which will be easier for client side to use.
+ */
+@Evolving
+public interface SupportsCatalogs {
```

Comment:
@shaofengshi I would suggest to move this client side interfaces to API module, and move the existing server side interfaces to the the core module, what do you think?

Reply:
@jerryshao done, moved the two to API module; and move the old version to the core module. All test passed.